Build Configuration & Application Installing

dylancopeland opened the pull request:
These commits have added build configuration and a new Vim command for installing a built application to a device. The changes are outlined in in `README.md` however I'll briefly outline them here:

**Build Configuration**
A `.build` file in a project's root directory will be used to pass switches to `xcodebuild` when building. I'm not sure if a "selected" configuration and scheme is stored anywhere in an `xcodeproj`, but if it is, I'll definitely look into using this instead.

**Installing**
Using `fruitstrap`, built applications can be installed to a device using `:Xinstall device`. Right now, only devices are supported but I'm going to add simulator support soon.

dylancopeland committed:
Added support for installing a built application onto a device. Updated the README accordingly.

dylancopeland committed:
Clarified a few things in the README and fixed a typo.

eraserhd commented:
Hi Dylan!

Thanks for submitting. I like where you're going with this. 

I also want to make sure the existing tests continue to pass and we maintain consistent code coverage. This pull request broke existing tests. Would you mind resubmitting, taking into account the existing tests and adding new specs for your new functionality?

Jason

dylancopeland committed:
Tests are passing now.

dylancopeland commented:
Hey Jason,

Sorry about that! I completely forgot to run the tests before pushing. Anyways, I have gone ahead and fixed my code and updated the tests accordingly. This is a great project that I'm looking forward to contributing to. :smile:

dylancopeland committed:
Build function now takes optional parameters.

eraserhd commented:
Can you add tests for the new features? :D

dylancopeland committed:
Added tests for installing app bundles.

dylancopeland commented:
I've added the tests. Let me know if anything else needs to be changed. :smiley:

eraserhd commented:
Thanks!
